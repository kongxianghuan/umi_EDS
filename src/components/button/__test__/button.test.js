import React from 'react'
import { shallow } from 'enzyme'
import Button from '..'

describe('Button', () => {
  it('renders primary style', () => {
    const button = shallow(<Button primary>Click</Button>)
    expect(button).toMatchSnapshot()
  })

  it('renders warning style', () => {
    const button = shallow(<Button warning>Click</Button>)
    expect(button).toMatchSnapshot()
  })
  
  it('renders with icon', () => {
    const button = shallow(<Button icon="icon-search">Click</Button>)
    expect(button).toMatchSnapshot()
  })
})