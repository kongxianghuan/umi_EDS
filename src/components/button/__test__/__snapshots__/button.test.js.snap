// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render correctly 1`] = `
<button
  className="btn  warning"
>
  <i
    className="icon icon-search"
  />
  Click
</button>
`;

exports[`Button renders primary style 1`] = `
<button
  className="btn primary "
>
  Click
</button>
`;

exports[`Button renders warning style 1`] = `
<button
  className="btn  warning"
>
  Click
</button>
`;

exports[`Button renders with icon 1`] = `
<button
  className="btn  "
>
  <i
    className="icon icon-search"
  />
  Click
</button>
`;
